### DISCUSSION TITLE
Idea: support resources in .zip format

### DISCUSSION THREAD

**Discussion by orent** [neutralinojs/neutralinojs#1175] [Ideas]
The resource file uses a custom .neu format. Using .zip would make it more flexible. Standard tools can be used to create and update it. This can make the neu utility unnecessary. And most importantly: zip supports appending to an executable. This can make it possible to pack a neutralino app as a single executable. 

https://libzip.org

**Comment by soshimee** [comment]
I agree, it would be a step forward to single-file executable.

**Comment by pegvin** [comment]
Actually there is nothing like `.neu` format, it's just what shalitha did. actually the file format of the `resource.neu` is "ASAR" file format. this is a type of archive just like zip. frameworks like Electron use the [ASAR](https://www.npmjs.com/package/asar) file format to pack and unpack files.

Even you can unpack the resource.neu file:

[![asciicast](https://asciinema.org/a/471209.svg)](https://asciinema.org/a/471209)

**Reply by hschneider** [reply]
I think a single file application is a rather seldom use-case. Most apps have more external dependencies, especially when extensions are used. So ASAR is IMHO a good solution.

**Reply by orent** [reply]
There’s not not much point in using less common formats. A .jar is actually a zip. Even a .docx file is actually a zip. It’s pretty much the industry default cross platform archive format.

**Reply by darthgelum** [reply]
Asar is used by Electron apps, like Visual Studio Code, for example. And you can open it with 7zip with a plugin, so it can be viewed and edited like any other archive

**Reply by hrgdavor** [reply]
but it is not compressed. I have an app with lots of documentation, and resources.neu is 56MB ... zipped it is 12MB. makes a big difference.

I will definitely need to explore other options
